src/pty.rs

```diff
@@ -52,15 +47,10 @@ pub fn ptsname_r(fd: RawFd) -> Result<String> {
         return Err(Error::last().into());
     }
 
-    let name_cstr = unsafe { CStr::from_ptr(name_buf.as_ptr()) };
-    let name = match name_cstr.to_str() {
-        Ok(s) => s.to_string(),
-        Err(_) => {
-            // FIXME: Unsure what error I should report here.
-            return Err(Error::last().into());
-        }
+    let name = unsafe {
+        CStr::from_ptr(name_buf.as_ptr())
     };
-    Ok(name)
+    Ok(name.to_string_lossy().into_owned())
```

Comment:
Yeah, I would think that `gethostname` should return a `Result<String>` instead of returning it in an output argument as a `CStr`. @posborne Any comments here?